generic_throttle/src/generic_throttle/generic_throttle.py

```diff
@@ -0,0 +1,151 @@
+
+
+class GenericThrottle:
+    def __init__(self):
+        self.topic_dictionary = None
+        self.parameters = {'namespace': None,
+                           'topic_framerate': None,
+                           'latched': False,
+                           'lazy': False}
+
+        rospy.init_node('generic_throttle')
+        rospy.on_shutdown(self._shutdown)
+
+        # Read mandatory /generic_throttle/* parameters from server
+        parameter_list = ['namespace', 'topic_framerate']
+        for parameter_name in parameter_list:
+            parameter_string = '/generic_throttle/' + parameter_name
+            if rospy.has_param(parameter_string):
+                self.parameters[parameter_name] = \
+                    rospy.get_param(parameter_string)
+            else:
+                rospy.logerr('Parameter ' + parameter_string + ' not available')
+                exit(5)
+
+        # Read optional /generic_throttle/* parameters from server
+        parameter_list = ['latched', 'lazy']
+        for parameter_name in parameter_list:
+            parameter_string = '/generic_throttle/' + parameter_name
+            if rospy.has_param(parameter_string):
+                self.parameters[parameter_name] = \
+                    rospy.get_param(parameter_string)
+
+        # Check if each entry of topic_framerate has 2 entries
+        size_flag = all(len(item) == 2 for item in
+                        self.parameters['topic_framerate'])
+
+        if(not(size_flag)):
+            rospy.logerr('Parameter /generic_throttle/framerate must be'
+                         ' a list of size 2 lists')
+            exit(10)
+
+        # Populate the dictionary for the ros topic throttling
+        self._populate_dictionary()
+        rospy.spin()
+
+    def timer_callback(self, event, topic_id):
+        # The False argument is for a non blocking call
+        locking = self.topic_dictionary[topic_id]['lock'].acquire_lock(False)
+
+
+        if not(locking):
+            current_t = currentThread()
+            rospy.logdebug(str(current_t._Thread__name) + ': cannot lock topic '
+                          + topic_id)
+            return
+
+        publisher = self.topic_dictionary[topic_id]['publisher']
+        subscriber = self.topic_dictionary[topic_id]['subscriber']
+
+        # Check if pub and sub already exist, if not try to create them after
+        # checking the ros topic
+
+        if None in(publisher, subscriber):
+            topic_info = rostopic.get_topic_class(topic_id)
+            if topic_info[0] is None:
+                rospy.logwarn('Cannot find topic ' + topic_id)
+
+                self.topic_dictionary[topic_id]['lock'].release_lock()
+                return
+            else:
+                # Create publisher
+                self.topic_dictionary[topic_id]['publisher'] = \
+                    rospy.Publisher(self.parameters['namespace'] + topic_id,
+                                    topic_info[0], queue_size=1)
```

Comment:
simply use the ```latched``` option of the Publisher constructor (see http://docs.ros.org/api/rospy/html/rospy.topics.Publisher-class.html#__init__) for achieving the latched behavior....no need to save the last message yourself....